EUR-Lex.js

```diff
@@ -105,112 +115,175 @@ function doWeb(doc, url) {
-	if (languageUrl=="AUTO") {
+	var languageUrl = url.split("/")[4].toUpperCase();
+	if (languageUrl == "AUTO") {
 		languageUrl = autoLanguage || "EN";
 	}
-	var language = languageMapping[languageUrl] || "eng";
-	
-	item.title = attr(doc, 'meta[property="eli:title"][lang=' + languageUrl.toLowerCase() + ']', 'content');
+	var language = languageMapping[languageUrl][0] || "eng";
+	// Cases only return language; discard everything else
 	item.language = languageUrl.toLowerCase();
 	
-	var uri = attr(doc, '#format_language_table_digital_sign_act_' + languageUrl.toUpperCase(), 'href');
-	if (uri) {
-		var uriParts = uri.split('/').pop().replace('?uri=', '').split(':');
-		// e.g. uriParts =  ["OJ", "L", "1995", "281", "TOC"]
-		// e.g. uriParts = ["DD", "03", "061", "TOC", "FI"]
-		if (uriParts.length>=4) {
-			if (/\d+/.test(uriParts[1])) {
-				item.code = uriParts[0];
-				item.codeNumber = uriParts[1] + ', ' + uriParts[2];
-			} else {
-				item.code = uriParts[0] + ' ' + uriParts[1];
-				item.codeNumber = uriParts[3];
-			}
-			if (type=="bill") {
-				item.codeVolume = item.code;
-				item.code = item.codeNumber;
+
+	if (eliTypeUri) {
+		// type: everything with ELI (see var typeMapping: bill, statute, report)
+		item.title = attr(doc, 'meta[property="eli:title"][lang=' + languageUrl.toLowerCase() + "]", "content");
+		var uri = attr(doc, "#format_language_table_digital_sign_act_" + languageUrl.toUpperCase(), "href");
+		if (uri) {
+			var uriParts = uri.split("/").pop().replace("?uri=", "").split(":");
+			// e.g. uriParts =  ["OJ", "L", "1995", "281", "TOC"]
+			// e.g. uriParts = ["DD", "03", "061", "TOC", "FI"]
+			if (uriParts.length >= 4) {
+				if (/\d+/.test(uriParts[1])) {
+					item.code = uriParts[0];
+					item.codeNumber = uriParts[1] + ", " + uriParts[2];
+				} 
+				else {
+					item.code = uriParts[0] + " " + uriParts[1];
+					item.codeNumber = uriParts[3];
+				}
+				if (type == "bill") {
+					item.codeVolume = item.codeNumber;
+					item.codeNumber = null;
+				}
 			}
 		}
+
+		// item.number = attr(doc, 'meta[property="eli:id_local"]', 'content');
+
+		item.date = attr(doc, 'meta[property="eli:date_document"]', "content");
+		if (!item.date) {item.date = attr(doc, 'meta[property="eli:date_publication"]', "content")}
+		
+		var passedBy = doc.querySelectorAll('meta[property="eli:passed_by"]');
+		var passedByArray = [];
+		for (let i = 0; i < passedBy.length; i++) {
+			passedByArray.push(passedBy[i].getAttribute("resource").split("/").pop());
+		}
+		item.legislativeBody = passedByArray.join(", ");
+		
+		item.url = attr(doc, 'meta[typeOf="eli:LegalResource"]', "about") + "/" + language.toLowerCase();
+		// eli:is_about -> eurovoc -> tags
 	}
-	
-	item.number = attr(doc, 'meta[property="eli:id_local"]', 'content');
-	
-	item.date = attr(doc, 'meta[property="eli:date_publication"]', 'content');
-	// attr(doc, 'meta[property="eli:date_document"]', 'content');
+	else if (docSector == "6") {
```

Comment:
If this branch only runs when `item.itemType == 'case'`, the `if` condition could be replaced with that for clarity.